### ISSUE TITLE
The pre-delivery gate false-fails on `.`, and blames the frontmatter for an argument problem

### ISSUE THREAD

**Issue by leobaray** [rebelytics/one-skill-to-rule-them-all#61]
# The pre-delivery gate false-fails on `.`, and blames the frontmatter for an argument problem

Verified against `release/v3.0.0` @ `433d31b` (not `main`, which is still 2.0.0 and has no `scripts/`).

## What happens

`scripts/validate-skill-bundle.py` builds its path without resolving it:

```python
# line 51
skill_dir = pathlib.Path(skill_dir)
```

and later compares the frontmatter `name` against `skill_dir.name` (line 75). For a
relative argument that names the current directory, `Path('.').name` is the empty
string, so a **correct** bundle fails:

```console
$ cd /path/to/staged/task-observer
$ python3 scripts/validate-skill-bundle.py .
warn: description 1014 chars (cap 1024) — near the boundary
FAIL:
  - frontmatter: `name` 'task-observer' != directory ''
$ echo $?
1
```

Same for `./`. The identical directory passes when addressed absolutely:

```console
$ python3 scripts/validate-skill-bundle.py /path/to/staged/task-observer
OK: all gate checks passed
```

The failure names `frontmatter:` as the failing check and prints an empty string as
the directory. Both halves point away from the actual cause, which is the argument.
The natural next move for anyone reading it is to go and inspect a `name:` field that
is already correct.

## Scope, stated honestly

`--pack` is correctly gated behind the directory checks, so this does **not** produce a
malformed bundle — the run refuses and writes nothing. The impact is a false failure on
a valid artefact plus a message that misdirects the reader. I mention the bound because
the fix is worth making for the message alone, not because the consequence is larger
than it is.

## Why the current text didn't prevent it

The gate landed in `311c7385` ("Pre-flight the literal command; make the delivery gate
assert"), whose message records: *"Pre-flighted against this skill's own bundle and
against a malformed one."* Both of those are content variations at a working path. #43's
rule — run the exact string from a clean shell — is about the *command*; nothing in it
asks for the *argument forms* the command will actually be given, and `.` is the form a
person produces after `cd`-ing into the directory they just staged.

This is the same class the gate exists to catch, one level up: #28 is about a gate that
measures without asserting; this is a gate that asserts against a value it never
normalised.

## Fix

```diff
--- a/scripts/validate-skill-bundle.py
+++ b/scripts/validate-skill-bundle.py
@@ -51 +51 @@
-    skill_dir = pathlib.Path(skill_dir)
+    skill_dir = pathlib.Path(skill_dir).resolve()
```

Verified after the change, from a clean shell, all three argument forms against the same
staged directory:

| argument | before | after |
|---|---|---|
| `.` | `FAIL: name 'task-observer' != directory ''` | `OK: all gate checks passed` |
| `./` | `FAIL: name 'task-observer' != directory ''` | `OK: all gate checks passed` |
| `/abs/path/task-observer` | `OK` | `OK` |

No regression on the check's actual purpose: a copy of the same bundle in a directory
named `nome-errado` still fails, before and after, with
`frontmatter: 'task-observer' != directory 'nome-errado'`.

Optionally, when this check fails, echo the argument as given alongside the resolved
basename — a gate's failure message should name the input it rejected.

## Not a duplicate

Searched all 46 issues and both open PRs. #28 is the closest (it asked for the gate) and
#43 is the rule the gate's commit implements; neither reports this. No issue or PR
mentions `validate-skill-bundle.py`'s argument handling.

Environment: Linux, Python 3.12, GNU coreutils. Run from a clean shell against a fresh
`git archive` of `release/v3.0.0`.

**Comment by rebelytics** [comment]
Fixed in v3.0.0, released today, exactly as you designed it: the path is resolved before the name comparison, and `.`, `./` and absolute invocations now agree (verified against your repro). The commit carries you as co-author. Thank you for a model bug report: verified against the right branch, minimal repro, root cause and fix included.
